packages/app-builder-lib/src/node-module-collector/pnpmNodeModulesCollector.ts

```diff
@@ -0,0 +1,25 @@
+import { execSync } from "child_process"
+import { NodeModulesCollector } from "./nodeModulesCollector"
+import { DependencyTree } from "./types"
+
+export class PnpmNodeModulesCollector extends NodeModulesCollector {
+  constructor(rootDir: string) {
+    super(rootDir)
+  }
+
+  getPMCommand(): string {
+    const cmd = process.platform === "win32" ? "pnpm.cmd" : "pnpm"
+    return `${cmd} list --prod --json --long --depth Infinity`
+  }
+
+  getDependenciesTree() {
+    const pnpmListOutput = execSync(this.getPMCommand(), {
+      cwd: this.rootDir,
+      encoding: "utf-8",
+      maxBuffer: 1024 * 1024 * 100,
```

Comment:
Nvm, it looks like the util.ts exec function uses 1000mb 😅. Might be worthwhile to leverage its implementation instead of calling `execSync` directly. Curious on your thoughts though
https://github.com/electron-userland/electron-builder/blob/74d98d8aaecf1f6ee2fbd7b639972d3eab1cd0c5/packages/builder-util/src/util.ts#L73